src/MSALWrapper/PCAWrapper.cs

```diff
@@ -61,6 +63,19 @@ public PCAWrapper(ILogger logger, IPublicClientApplication pca, IList<Exception>
         /// </summary>
         public bool UseEmbeddedWebView { get; private set; } = false;
 
+        /// <summary>
+        /// Gets the cache file name. Only available on Windows.
+        /// </summary>
+        /// <param name="tenantId">The tenant ID.</param>
+        /// <returns>The absolute path for cache file corresponding to the tenant.</returns>
+        public static string GetCacheFilePath(Guid tenantId)
+        {
+            // Use default cache file path.
+            string appData = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+            string absolutePath = Path.Combine(appData, ".IdentityService", $"msal_{tenantId}.cache");
+            return absolutePath;

```

Comment:
```suggestion
            return Path.Combine(Environment.SpecialFolder.LocalApplicationData.ToString(), ".IdentityService", $"msal_{tenantId}.cache");

```

Reply:
This is where the bug actually was - the `Enviornment.GetFolderPath` method was returning null if the path doesn't exist. Technically this could happen on Windows too if someone deleted their AppData folder. We can remove that call because the constant always returns the actual value.

Reply:
I tried it before.
`Environment.SpecialFolder.LocalApplicationData` is an enum type `SpecialFolder`. Use `ToString()` will return `LocalApplicationData`.